core/src/main/java/hudson/model/Job.java

```diff
@@ -206,20 +202,19 @@ public synchronized void save() throws IOException {
         holdOffBuildUntilSave = holdOffBuildUntilUserSave;
     }
 
-    @Override public void onCreatedFromScratch() {
-        super.onCreatedFromScratch();
-        runIdMigrator = new RunIdMigrator();
-        runIdMigrator.created(getBuildDir());
-    }
-
     @Override
     public void onLoad(ItemGroup<? extends Item> parent, String name)
             throws IOException {
         super.onLoad(parent, name);
 
-        File buildDir = getBuildDir();
-        runIdMigrator = new RunIdMigrator();
-        runIdMigrator.migrate(buildDir, Jenkins.get().getRootDir());
+        // see https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449
+        // This code can be deleted after several Jenkins releases,
+        // when it is likely that everyone is running a version equal or higher to this version.
+        var buildDirPath = getBuildDir().toPath();
+        if (Files.deleteIfExists(buildDirPath.resolve("legacyIds"))) {
+            LOGGER.info("Deleting legacyIds file in " + buildDirPath + ". See https://issues.jenkins"
+                        + ".io/browse/JENKINS-75465 for more information.");
+        }
```

Comment:
Continuing from [this comment](https://github.com/jenkinsci/jenkins/pull/10456#issuecomment-2748112449):

[JENKINS-64356](https://issues.jenkins.io/browse/JENKINS-64356) discusses slow startup caused by a missing legacyIds file. This appears to happen because runIdMigrator.migrate(buildDir, Jenkins.get().getRootDir()) iterates through all builds to check for necessary migrations.

CloudBees provided a [KB article](https://docs.cloudbees.com/docs/cloudbees-ci-kb/latest/client-and-managed-controllers/jenkins-startup-logs-show-runidmigrator-logs#_workaround) with a workaround to recreate the legacyIds file when it's missing.

Since the `legacyIds` file is deleted here, downgrading to an earlier version could result in slow startup.

Reply:
I wrote a note in `changelog` and `upgrade guidelines` to mention about downgrading.